src/components/NewCommentForm.tsx

```diff
@@ -1,37 +1,196 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { CommentData } from '../types/Comment';
+import { ErrorMessage } from './ErrorMessage';
+
+interface ErrorMessages {
+  nameError: string,
+  emailError: string,
+  bodyError: string,
+  serverError: string,
+}
+
+interface Props {
+  handleAddNewComment: (commentData: CommentData) => Promise<void>,
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  handleAddNewComment,
+}) => {
+  const [errors, setErrors] = useState<ErrorMessages>({
+    nameError: '',
+    emailError: '',
+    bodyError: '',
+    serverError: '',
+  });
+
+  const [commentData, setCommentData] = useState<CommentData>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleReset = () => {
+    setCommentData({
+      name: '',
+      email: '',
+      body: '',
```

Comment:
I would create const for that values outside component like `initialValues` or `defaultValues `for 

```
{
  nameError: '',
  emailError: '',
  bodyError: '',
  serverError: '',
}
```

```
{
  name: '',
  email: '',
  body: '',
}
  ```
  
and reuse everywhere you need in your file